### PR TITLE
Resolution for issue  #757

### LINKED ISSUE #757: ColourCodes doesn't handle rgba !
https://duckduckgo.com/?q=rgba%2899%2C60%2C176%2C0.5%29
 vs
https://duckduckgo.com/?q=rgb%2899%2C60%2C176%29

https://duck.co/ia/view/color_codes

### PR DESCRIPTION
**Pull request by mintsoft**
Not the prettiest implementation, however it does work. Any refactoring ideas welcomed.

fixes #757

### CODE REVIEW

**Comment by jagtalon** [comment]
Hey, @mintsoft!

- Looks like there's too much space at the bottom:
screen shot 2015-01-09 at 3 00 28 pm
- The links "Images" and "Info" should be grey (maybe the class `tx-clr--lt`)
- The rgba, hsl, etc. should be in all caps like the one in the mockup:

screen shot 2015-01-09 at 3 04 00 pm


color-hex1

File: lib/DDG/Goodie/ColorCodes.pm [hunk 1]
```diff
@@ -1,14 +1,16 @@
 package DDG::Goodie::ColorCodes;
-# ABSTRACT: Copious information about various ways ofencoding colors.
+# ABSTRACT: Copious information about various ways of [encoding colors.
```
Comment (jagtalon) [inline lib/DDG/Goodie/ColorCodes.pm]: Looks like a typo "of [encoding"

**Comment by mintsoft** [comment]
@jagtalon I'll make those changes shortly; however I deliberately changed the format around the RGB values etc so that they're copy-pastable for use in CSS (personally; my primary use-case for this goodie!). I can't really think of another way of satisfying both 'needs' as it were

**Comment by chrismorast** [comment]
Thanks @mintsoft , I apologize for the delay in getting back to you on this. This is really close. I Will put together some minor css updates for you and post sometime on Monday.

**Comment by mintsoft** [comment]
No problem @chrismorast

**Comment by chrismorast** [comment]
Here's the feedback (in addition to @jagtalon 's from above)...

I'm attaching a screenshot with some guidelines and alignment reference.
screen shot 2015-01-13 at 11 44 12 am

screen shot 2015-01-13 at 11 46 25 am

And the comments to reference:
1. Please vertically center the grouped content with the associated swatch(s) 
2. The 2 sections on the right (complimentary and analogous) should have a gutter between them so the spacing between the title and the values is smaller than the gutter (as shown in the screenshot but the pink and blue blocks of color). This will alllow the 2 groups to be visually separated)
3. Please refer to the guidelines in the screenshot for grouping alignment (baseline of the analogous value text should match the baseline of the links and the top of the 'complimentary' label should be aligned with the top of the 'rgb' values
4. Add a space between the 'main info labels (i.e.. rgb, hsl, & cmyk) and their values
5. Text colors should be something like this...
screen shot 2015-01-13 at 11 55 01 am

I'm not sure how to address the text colors to allow proper updating for theme changes so maybe @jagtalon can help us out there?

**Comment by mintsoft** [comment]
@jagtalon @chrismorast @moollaza any thoughts about the copy-paste availability of the result for css purposes etc?

**Comment by chrismorast** [comment]
Sorry @mintsoft , I'm okay with giving that a shot. I was just using the old comp in my screenshots above.

**Comment by jagtalon** [comment]
@chrismorast @mintsoft you can use `tx-clr--dk2` for the `#666` color and `tx-clr--dk` for `#333`. It should adjust the color for you if you just use these classes for coloring your text.

**Comment by mintsoft** [comment]
cool will do

**Comment by mintsoft** [comment]
@jagtalon @chrismorast Sorry about the delay; I totally forgot about this PR! The alignments should be consistent now

**Comment by jagtalon** [comment]
@mintsoft thanks--putting it up on a server!

**Comment by moollaza** [comment]
@jagtalon put it on a server for us: https://ddh1.duckduckgo.com/?q=hex+color+code+for+cyan&ia=answer

@mintsoft @jagtalon @chrismorast this looks awesome! Nicely done team.

The only thing I noticed is that the "Images" and "Info" links seem to be broken?

https://ddh1.duckduckgo.com/?q=%23123456&ia=answer

**Comment by jagtalon** [comment]
Ah nice catch @moollaza!

**Comment by mintsoft** [comment]
@moollaza well spotted!

@jagtalon should be fixed now

**Comment by mintsoft** [comment]
@moollaza @jagtalon is this ready to `$rock && $roll`?

**Comment by jagtalon** [comment]
Doing some final checks!

**Comment by chrismorast** [comment]
Hey @mintsoft , here are some final design tweaks:
1. Vertically center the content group with the main swatch (refer to screenshot)
2. Left align the main swatch with the rest of the SERP content (organics etc - refer to screenshot)
3. 'Images' link doesn't take you to the correct page (an empty grid) 
4. Can we change the separator between the links to #d0d0d0
5. Can we make the hex values in the complimentary and analogous sections clickable (for a requery)?

I also did some testing on the data and noticed some discrepancies; 
1. The 2nd analogous color is not consistent with what I'm seeing on the sources site
screen shot 2015-02-04 at 4 46 41 pm
2. The CMYK and HSL values don't produce the given color. I've pulled up the photoshop color picker and you can see the discrepancies here.
screen shot 2015-01-30 at 1 46 49 pm

**Comment by mintsoft** [comment]
@chrismorast I've updated everything except for the issue around the HSL & CMYK values; they appear to be consistent between the goodie (https://duckduckgo.com/?q=%23123456&t=ffab&ia=answer) and color-hex (http://www.color-hex.com/color/123456)

I've not actually changed any of the handling around HSL or CMYK so I've been presuming they're correct.

For example for #123456:

HSL	0.58	0.65	0.20 (which is 210,65%, 20%)
CMYK 	0.79	0.40	0.00   0.66

I'm not really sure where to go from here, any ideas? @moollaza @chrismorast

**Comment by chrismorast** [comment]
Awesome, thanks @mintsoft ! I'll look into the CMYK and HSL values a little further as well. Maybe I'm missing something there.

**Comment by chrismorast** [comment]
@mintsoft , I apologize for the delay in getting back to you here. I did some research and it seems this is accurate (although not producing accurate colors). I tested several conversion tools and got the same values though. 

Thank you for all of your work on this, this is ready to merge!

**Comment by moollaza** [comment]
@chrismorast awesome, thanks!

@mintsoft mind rebasing this? and then I'll gladly merge it in.

**Comment by mintsoft** [comment]
@moollaza rebased

**Comment by moollaza** [comment]
@chrismorast @mintsoft this looks awesome. There's a couple edge cases I found, but I think we can tackle them when we convert this to a custom template (which will hopefully be soon!)

Specifically if someone searches for the colour that we're using as the background for the Goodie div:

2015-02-24_20h32_10

Also, some colors like #FFF and #000 don't have analogous/complementary colors so we just show the same colour 3 more time which I think is OK?

2015-02-24_20h36_00

**Comment by moollaza** [comment]
Just noticed that our link font colour doesn't work well on the dark theme -- our CSS is setting it to #333 but it would otherwise be #FFF because of the `tx-clr-dk` class...

**Comment by chrismorast** [comment]
Can we try a mostly transparent inner border similar to what we do for the team section of the about page? Maybe just go 1px with it on the small circles though.
screen shot 2015-02-24 at 8 04 20 pm

**Comment by mintsoft** [comment]
@moollaza @chrismorast The border is relatively easy to fix (I think probably 1px solid black border will do it).

Is there a class that's better placed for the link colours so it doesn't break with the dark theme?

**Comment by mintsoft** [comment]
@moollaza @chrismorast circle borders added, just the colours in dark to solve now.
image

**Comment by moollaza** [comment]
@mintsoft nicely done and looking good. I'll merge this in and find out about the font class for the links.

**Comment by moollaza** [comment]
@mintsoft after a little more digging I see we have an alternate class, `tx--clr--dk2` -- but we give it the same color as `tx--clr--dk` when we switch the the dark theme so it doesn't make a difference.

@chrismorast any thoughts? Shall we just specify a 2nd colour for the dark theme? (seems like this might be a problem for us to fix in our CSS potentially)

**Comment by chrismorast** [comment]
Thanks @mintsoft !
Regarding the links, I believe we already have a solution for this. cc @sdougbrown @bsstoner .

**Comment by mintsoft** [comment]
@moollaza @sdougbrown @bsstoner Sorry to pester; but it'd be nice to get this locked down!

**Comment by moollaza** [comment]
@mintsoft apologies. This is at the 99% point, I'll merge and set it up for release. We should be able to figure out fonts soon enough.

**Comment by mintsoft** [comment]
Cool! Thanks @moollaza Do you think we should have an issue open for the font classes?

**Comment by moollaza** [comment]
@mintsoft it's an internal issue so I'll talk with the team offline. We can certainly create an issue to keep the community informed.

**Pull request merged**